WalletWasabi/Tor/Http/IHttpClient.cs

```diff
@@ -15,5 +17,51 @@ public interface IHttpClient
 
 		/// <inheritdoc cref="HttpClient.SendAsync(HttpRequestMessage, CancellationToken)"/>
 		Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken token = default);
+
+		Task<HttpResponseMessage> SendAsync(HttpMethod method, Uri requestUri, HttpContent? content = null, CancellationToken token = default)
+		{
+			var httpRequestMessage = new HttpRequestMessage(method, requestUri);
+
+			if (content is { })
+			{
+				httpRequestMessage.Content = content;
+			}
+
+			return SendAsync(httpRequestMessage);
+		}
+
+		/// <summary>
+		/// Sends HTTP request (up to <paramref name="retry"/> times) to get an HTTP response.
+		/// </summary>
+		/// <param name="method">HTTP method to use to send the HTTP request.</param>
+		/// <param name="requestUri">URI to send HTTP request to. Cannot be relative URI.</param>
+		/// <param name="retry">Maximum number of attempts to do to get HTTP response.</param>
+		/// <exception cref="OperationCanceledException">When <paramref name="token"/> is used to cancel operation.</exception>
+		public async Task<HttpResponseMessage?> SendAndRetryAsync(HttpMethod method, Uri requestUri, int retry = 2, HttpContent? content = null, CancellationToken token = default)
```

Comment:
Isn't this duplicated code?

Reply:
It's almost the same but it uses absolute URI. It is explained in PR's notes. It's not great but it's a path forward in my opinion because we may migrate to use absolute URIs which would make TorHttpClient and similar classes state-less classes which would make it easier to create a pool of these clients to avoid those async locks in TorHttpClient.

Reply:
ACK